app/forms/search_form.rb

```diff
@@ -66,6 +66,10 @@ def remote_advice_method_ids
     advice_methods.select(&:present?)
   end
 
+  def advice_methods
+    @advice_methods.try(:map) { |e| e.to_i if e.present? }
```

Comment:
compact doesn't fix the problem - I need to get rid of empty strings as well...

Do you like current implementation?